Files/Helpers/JumpListManager.cs

```diff
@@ -0,0 +1,114 @@
+        private JumpList _instance = null;
+        private List<string> JumpListItemPaths { get; set; }
+        public JumpListManager()
+        {
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            if (JumpList.IsSupported())
+            {
+                _instance = await JumpList.LoadCurrentAsync();
+
+                // Disable automatic jumplist. It doesn't work with Files UWP.
+                _instance.SystemGroupKind = JumpListSystemGroupKind.None;
+                JumpListItemPaths = _instance.Items.Select(item => item.Arguments).ToList();
+            }
+        }
+
+        public async void AddFolderToJumpList(string path)
+        {
+            await AddFolder(path);
+            await _instance?.SaveAsync();
+        }
+
+        private Task AddFolder(string path)
+        {
+            if (!JumpListItemPaths.Contains(path) && _instance != null)
+            {
+                string displayName;
+                if (path.Equals(App.AppSettings.DesktopPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarDesktop";
+                }
+                else if (path.Equals(App.AppSettings.DownloadsPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarDownloads";
+                }
+                else if (path.Equals(App.AppSettings.DocumentsPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarDocuments";
+                }
+                else if (path.Equals(App.AppSettings.PicturesPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarPictures";
+                }
+                else if (path.Equals(App.AppSettings.MusicPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarMusic";
+                }
+                else if (path.Equals(App.AppSettings.VideosPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarVideos";
+                }
+                else if (path.Equals(App.AppSettings.RecycleBinPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    var localSettings = ApplicationData.Current.LocalSettings;
+                    displayName = localSettings.Values.Get("RecycleBin_Title", "Recycle Bin");
+                }
+                else
+                {
+                    displayName = Path.GetFileName(path);
+                }
+
+                var jumplistItem = JumpListItem.CreateWithArguments(path, displayName);
+                jumplistItem.Description = jumplistItem.Arguments;
+                jumplistItem.GroupName = "ms-resource:///Resources/JumpListRecentGroupHeader";
```

Comment:
You don't need to translate group name as it should be translated by default. Look at this [link](https://github.com/microsoft/Windows-universal-samples/blob/a58c5d9b84456196889edb1f4a6e844c7a729b8b/Samples/JumpList/cs/Scenario4_ChangeSystemGroup.xaml.cs#L49).

Reply:
This is for the automatic recent list which doesn't work with Files UWP since this app isn't not associated with any file type.